test/unit/models/users.test.js

```diff
@@ -474,4 +477,74 @@ describe("users", function () {
       expect(userListResult[0].discordId).to.be.deep.equal(userDataArray[0].discordId);
     });
   });
+  describe("Adding github_user_id for each user document", function () {
+    let userId, userToken, superUserId, superUserToken;
+    beforeEach(async function () {
+      userId = await addUser(prodUsers[1]);
+      userToken = authService.generateAuthToken({ userId: userId });
+      superUserId = await addUser(prodUsers[0]);
+      superUserToken = authService.generateAuthToken({ userId: superUserId });
+    });
+
+    afterEach(async function () {
+      await cleanDb();
+    });
+
+    it("API should not be accessible to any regular user", function (done) {
+      chai
+        .request(app)
+        .post("/users/migrations?action=adds-github-id&skip=0&limit=10")
+        .set("cookie", `${cookieName}=${userToken}`)
+        .send()
+        .end((err, res) => {
+          if (err) {
+            return done();
+          }
+          expect(res).to.have.status(401);
+          expect(res.body).to.be.an("object");
+          expect(res.body).to.eql({
+            statusCode: 401,
+            error: "Unauthorized",
+            message: "You are not authorized for this action.",
+          });
+          return done();
+        });
+    });
+    it("API should be accessible by super user", async function () {
+      for (const user of prodUsers.slice(2)) {
+        await addUser(user);
+      }
+      const allUsers = await chai.request(app).get("/users").set("cookie", `${cookieName}=${superUserToken}`).send();
+      const usersWithoutGithubId = allUsers.body.users.filter((user) => {
+        return !user.github_user_id;
+      });
+
+      expect(usersWithoutGithubId).to.not.have.length(0);
+
+      const res = await chai
+        .request(app)
+        .post("/users/migrations?action=adds-github-id&skip=0&limit=10")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .send();
+
+      expect(res).to.have.status(200);
+      expect(res.body).to.have.property("data").that.is.an("object");
+      expect(res.body.data).to.have.property("totalUsers").that.is.a("number");
+      expect(res.body.data).to.have.property("usersUpdated").that.is.a("number");
+      expect(res.body.data).to.have.property("usersNotUpdated").that.is.a("number");
+      expect(res.body.data).to.have.property("invalidUsersDetails").that.is.an("array");
+
+      const updatedUsers = await chai
+        .request(app)
+        .get("/users")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .send();
+
+      const updatedUsersWithoutGithubId = updatedUsers.body.users.filter((user) => {
+        return !user.github_user_id;
+      });
+      // For invalid username
+      expect(updatedUsersWithoutGithubId).to.have.length(1);
```

Comment:
What does this test actually do? 
- Name suggests that it only checks if the route is accessible by super user. 
- What are these  checks? can we move them to separate test with appropriate name.

Reply:
Done